### PR TITLE
Copter: add SAFETY_STATUS for Prearm

### PR DESCRIPTION
**Pull request by SkiTiSu**
Hi everyone, this is the achievement for issue #2171.

Because my level is limited, maybe there're some mistakes or inadequacies. Please give me a hand.

By the way, the code was the version of over 20 days before when I edited, so I didn't notice there's a new check for "Motor Interlock", I will add it later.

If my method is feasible, I'll creat the UI in MP.

Thanks.

### CODE REVIEW

**Comment by SkiTiSu** [comment]
meee1 suggested that this message can tie with the EKF status. I think it's a good idea.
Any advice?

**Comment by magicrub** [comment]
@rmackay9 is working on this in parallel. You guys should talk.

**Comment by SkiTiSu** [comment]
@rmackay9 Hmmm, it have conflicts now :(

**Comment by magicrub** [comment]
This needs a rebase on master due to the namespace change of HAL to access hal.*

File: ArduCopter/motors.pde [hunk 1]
```diff
@@ -288,19 +292,28 @@ static bool pre_arm_checks(bool display_failure)
...
-            gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: RC not calibrated"));
+        prearmstatus += PREARM_RC;
+        if (flag) {
+            flag = false;
+            if (display_failure) {
+                gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: RC not calibrated"));
+            }
         }
-        return false;
+        //return false;
```
Comment (magicrub) [inline ArduCopter/motors.pde]: please remove instead of commenting out code

File: ArduCopter/motors.pde [hunk 2]
```diff
@@ -288,19 +292,28 @@ static bool pre_arm_checks(bool display_failure)
...
-                gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Barometer not healthy"));
+            prearmstatus += PREARM_BARO;
+            if (flag) {
+                flag = false;
+                if (display_failure) {
+                    gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Barometer not healthy"));       
+                }
             }
-            return false;
+            //return false;
```
Comment (magicrub) [inline ArduCopter/motors.pde]: please remove instead of commenting out code

File: ArduCopter/motors.pde [hunk 3]
```diff
@@ -309,10 +322,14 @@ static bool pre_arm_checks(bool display_failure)
...
-                    gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Altitude disparity"));
+                prearmstatus += PREARM_ALT;
+                if (flag) {
+                    flag = false;
+                    if (display_failure) {
+                        gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Altitude disparity"));
+                    }
                 }
-                return false;
+                //return false;
```
Comment (magicrub) [inline ArduCopter/motors.pde]: please remove instead of commenting out code

File: ArduCopter/motors.pde [hunk 4]
```diff
@@ -321,36 +338,51 @@ static bool pre_arm_checks(bool display_failure)
...
-                gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Compass not calibrated"));
+            prearmstatus += PREARM_COMPASS_CALIBRATED;
+            if (flag) {
+                flag = false;
+                if (display_failure) {
+                    gcs_send_text_P(SEVERITY_HIGH,PSTR("PreArm: Compass not calibrated"));
+                }
             }
-            return false;
+            //return false;
```
Comment (magicrub) [inline ArduCopter/motors.pde]: please remove instead of commenting out code

**Comment by magicrub** [comment]
Hi @SkiTiSu 
This PR needs some work. Please take a look at our guidelines here http://dev.ardupilot.com/wiki/submitting-patches-back-to-master/
- commits need to be broken up by library directory
- for mavlink msg commits we like the xml change as a single commit, then the rest of the generated stuff as another commit.
- do not leave commented code in there. That's what version control is for! :)
Have you talked to @rmackay9 about this? At one point he was working on something similar.

**Comment by SkiTiSu** [comment]
Hi @magicrub 
Thank you for your information. I'm sorry I have recently busy. Maybe I could improve this PR some days later.

**Comment by R-Lefebvre** [comment]
I wonder how relevant this is anymore with Randy's recent work on AP_Arming?

@rmackay9 ?

**Comment by davidbuzz** [comment]
This PR is very old ( about a year) , we think it's probably no longer relevant so we are closing it now, please let us know if this is in error, or raise a new PR when you have an improved or updated version of the code available.

**Pull request closed without merging**